sqlglot/generator.py

```diff
@@ -4656,22 +4656,52 @@ def unpivotcolumns_sql(self, expression: exp.UnpivotColumns) -> str:
 
         return f"NAME {name} VALUE {values}"
 
-    def computestatistics_sql(self, expression: exp.ComputeStatistics) -> str:
+    def statistics_sql(self, expression: exp.Statistics) -> str:
+        kind = self.sql(expression, "kind")
+        option = self.sql(expression, "option")
+        option = f" {option}" if option else ""
         this = self.sql(expression, "this")
+        this = f" {this}" if this else ""
         columns = self.expressions(expression)
         columns = f" {columns}" if columns else ""
-        return f"COMPUTE STATISTICS {this}{columns}"
+        return f"{kind}{option} STATISTICS{this}{columns}"
+
+    def histogram_sql(self, expression: exp.Histogram) -> str:
+        this = self.sql(expression, "this")
+        columns = self.expressions(expression)
+        inner_expression = self.sql(expression, "expression")
+        inner_expression = f" {inner_expression}" if inner_expression else ""
+        update_options = self.sql(expression, "update_options")
+        update_options = f" {update_options} UPDATE" if update_options else ""
+        return f"{this} HISTOGRAM ON {columns}{inner_expression}{update_options}"
+
+    def usingdata_sql(self, expression: exp.UsingData) -> str:
+        data = self.sql(expression, "this")
+        return f"USING DATA {data}"
+
+    def analyzecolumns_sql(self, expression: exp.AnalyzeColumns) -> str:
+        return self.sql(expression, "this")
+
+    def analyzewith_sql(self, expression: exp.AnalyzeColumns) -> str:
+        expressions = " WITH ".join(expression.args.get("expressions") or [])
+        return f"WITH {expressions}" if expressions else ""
```

Comment:
I think something like this should work?
```suggestion
    def analyzewith_sql(self, expression: exp.AnalyzeColumns) -> str:
        return self.expressions(expression, prefix="WITH", sep=" ")
```